### PR TITLE
Remove fastjson library

### PR DESCRIPTION
**Pull request by gabrywu**
fix https://github.com/apache/beam/issues/24154, 
hi @apilloud can you help to review this PR?

### CODE REVIEW

**Comment by Abacn** [comment]
Thanks for working on this! There are merge conflicts - could you please rebase onto the current master? Also, consider add this change to CHANGES.md.

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryTableProviderTest.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public void testDefaultMethod_whenPropertiesAreNotSet() {
   public void testSelectDefaultMethodExplicitly() {
     Table table =
         fakeTableWithProperties(
-            "hello", "{ " + METHOD_PROPERTY + ": " + "\"" + Method.DEFAULT.toString() + "\" }");
+            "hello", "{\"" + METHOD_PROPERTY + "\": " + "\"" + Method.DEFAULT.toString() + "\" }");
```
Comment (Abacn) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryTableProviderTest.java:70]: there are multiple changes of adding quotes here. Do they mean a breaking change (SQL statements worked before now need to add quotes) / is there any chance keep the original semantics?

Comment (gabrywu) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryTableProviderTest.java:70]: I will fix it, with JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES true

**Comment by gabrywu** [comment]
> Thanks for working on this! There are merge conflicts - could you please rebase onto the current master? Also, consider add this change to CHANGES.md.

added

File: CHANGES.md [hunk 2]
```diff
@@ -89,6 +89,7 @@
 
 * Python SDK: Legacy runner support removed from Dataflow, all pipelines must use runner v2.
 * [Python] Dataflow Runner will no longer stage Beam SDK from PyPI in the `--staging_location` at pipeline submission. Custom container images that are not based on Beam's default image must include Apache Beam installation.([#26996](https://github.com/apache/beam/issues/26996))
+* SDK Java Extensions SQL: remove fastjson library, table property is changed to ObjectNode which belongs to jackson library. ([#24154](https://github.com/apache/beam/issues/24154))
```
Comment (Abacn) [inline CHANGES.md:92]: This will actually go to 2.51.0, fine for now, could fix later as part of adding 2.51.0 section

**Review by Abacn — APPROVED** [review]

**Comment by Abacn** [comment]
Run SQL PostCommit

**Comment by Abacn** [comment]
There are test failures:

```
org.apache.beam.sdk.extensions.sql.meta.provider.kafka.KafkaTableProviderIT.testFake2[0]

Error Message
java.lang.RuntimeException: illegal table properties: { "format" : "json",}
	at org.apache.beam.sdk.extensions.sql.TableUtils.parseProperties(TableUtils.java:50)
	at org.apache.beam.sdk.extensions.sql.meta.provider.kafka.KafkaTableProviderIT.testFake2(KafkaTableProviderIT.java:156)
...
```

Previously a ill-formed json `{ "format" : "json",}` works, now it does not. 

Similarly, it is nice to tolerant trailing comma

**Comment by gabrywu** [comment]
> 

I will add `ALLOW_TRAILING_COMMA` flag

File: CHANGES.md (hunk 2 above)
Comment (gabrywu) [reply CHANGES.md:92]: OK

**Comment by Abacn** [comment]
Thanks again for the fix. It appears fastjson parser in general is more relax than the default jackson mapper. Let me do some small experiments regarding the default behavior of two parsers, specifically, the list in https://fasterxml.github.io/jackson-core/javadoc/2.9/com/fasterxml/jackson/core/JsonParser.Feature.html

**Comment by Abacn** [comment]
Experiment result shows fastjson in general is much more relax in non-standard syntax. We may need to add other configs, see below.

| config | fastjson default | jackson default | jackson relaxed |
|---|---|---|---|
| normal | o | o | o |
| ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER | o | x | o |
| ALLOW_COMMENTS | o | x | o |
| ALLOW_MISSING_VALUES | o | x | o |
| ALLOW_NON_NUMERIC_NUMBERS | o | x | o |
| ALLOW_NUMERIC_LEADING_ZEROS | o | x | o |
| ALLOW_SINGLE_QUOTES | o | x | o |
| ALLOW_TRAILING_COMMA | o | x | o |
| ALLOW_UNQUOTED_CONTROL_CHARS | o | x | o |
| ALLOW_UNQUOTED_FIELD_NAMES | o | x | o |
| ALLOW_YAML_COMMENTS | x | x | o |

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/TableUtils.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.Map;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.annotations.VisibleForTesting;
+
+public class TableUtils {
+  private static ObjectMapper objectMapper =
+      new ObjectMapper()
+          .configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true)
+          .configure(JsonParser.Feature.ALLOW_TRAILING_COMMA, true);
```
Comment (Abacn) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/TableUtils.java:33]:
```suggestion
  private static ObjectMapper objectMapper =
      sonMapper.builder().enable(
JsonReadFeature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER,
        JsonReadFeature.ALLOW_JAVA_COMMENTS,
        JsonReadFeature.ALLOW_MISSING_VALUES,
        JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS,
        JsonReadFeature.ALLOW_LEADING_ZEROS_FOR_NUMBERS,
        JsonReadFeature.ALLOW_SINGLE_QUOTES,
        JsonReadFeature.ALLOW_TRAILING_COMMA,
        JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS,
        JsonReadFeature.ALLOW_UNQUOTED_FIELD_NAMES)
        .build();
```

**Comment by Abacn** [comment]
Run SQL PostCommit

**Pull request merged**